storage/interface.go

```diff
@@ -107,6 +121,17 @@ type LabelQuerier interface {
 	Close() error
 }
 
+type ExemplarQueryable interface {
+	// Querier returns a new ExemplarQuerier on the storage.
+	ExemplarQuerier(ctx context.Context) (ExemplarQuerier, error)
+}
+
+// Querier provides reading access to time series data.
+type ExemplarQuerier interface {
+	// Select returns a set of exemplars that have the given labels for the series they are associated with.
+	Select(start, end int64, l labels.Labels) ([]exemplar.Exemplar, error)
```

Comment:
I was expecting `ExemplarQuerier.Select` to take a set of matchers, not a set of labels.  I don't understand how we could make this fast in the future - this API doesn't allow for us to use the index, even if right now we brute force the search.